### PR TITLE
Implement automatic fever activation

### PR DESCRIPTION
**Pull request by LumpBloom7**
This PR introduces an option to enable automatic fever activation, which is on by default.

When it is on, fever will be activated when a result causes the progress to fill to 100%, using the `TryActivateFever` method.

`ReplayButtonState.Smoke` has been repurposed to store the auto fever setting, and is used to ensure that replays will use the settings set at the time of recording, so as to prevent desyncs. The `UsingAutoFever` state provided by `RushFramedInputHandler` will always take priority if a replay is active.

`FeverProcessor` uses `DrawableRushRuleset.UsingAutoFever` in order to determine whether it should automatically trigger fever, and has no knowledge of whether a replay is active or not. (`DrawableRushRulset.UsingAutoFever` already prioritizes replays)

### CODE REVIEW

File: osu.Game.Rulesets.Rush/Configurations/RushRulesetConfigManager.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+        protected override void InitialiseDefaults()
+        {
+            base.InitialiseDefaults();
+
+            SetDefault(RushRulsetSettings.AutomaticFever, true);
+        }
+    }
+
+    public enum RushRulsetSettings
```
Comment (swoolcock) [inline osu.Game.Rulesets.Rush/Configurations/RushRulesetConfigManager.cs:24]: Typo.

File: osu.Game.Rulesets.Rush/Replays/RushReplayFrame.cs [hunk 2]
```diff
@@ -47,6 +54,8 @@ public void FromLegacy(LegacyReplayFrame currentFrame, IBeatmap beatmap, ReplayF
                 ++currentBit;
                 flags >>= 1;
             }
+
+            UsingAutoFever = currentFrame.ButtonState.HasFlagFast(ReplayButtonState.Smoke);
```
Comment (swoolcock) [inline osu.Game.Rulesets.Rush/Replays/RushReplayFrame.cs:58]: Would appreciate a comment here regarding the repurposing of Smoke.

File: osu.Game.Rulesets.Rush/UI/RushSettingsSubsection.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            : base(ruleset)
+        {
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            var config = (RushRulesetConfigManager)Config;
+
+            // for an odd reason, Config seems to be passed as null when creating it. doesnt even get called...
```
Comment (swoolcock) [inline osu.Game.Rulesets.Rush/UI/RushSettingsSubsection.cs:25]: Elaborate?

**Review by swoolcock — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Rush/UI/RushSettingsSubsection.cs (hunk 3 above)
Comment (LumpBloom7) [reply osu.Game.Rulesets.Rush/UI/RushSettingsSubsection.cs:25]: I have no idea, this seems to be something that was a thing in the past, and was copied over without much thought.

It doesn't seem to be needed at all, will remove.

**Review by swoolcock — APPROVED** [review]
Looks good.

**Review by frenzibyte — APPROVED** [review]
The general direction and structure looks pretty good :+1:

I've applied a few cleanup changes and declared a `FeverActivationMode` enum for this, please give them a check and merge the PR if nothing looks off to you.

**Pull request merged**